### PR TITLE
Biot rhs

### PR DESCRIPTION
**Pull request by IvarStefansson**
Important note:
The time step was removed from MassMatrix.

Otherwise, there are few updates in fv/biot.py:
DivD class returns the rhs
Fixed two bugs in the Biot rhs

utils/test_biot.py was extended with a test of the rhs. See that file for how to assemble Biot with a non-zero rhs. In particular, the assemble_rhs functions of the classes Mpfa and MassMatrix must be adjusted to account for the time step and the rhs, respectively.

### CODE REVIEW

File: src/porepy/numerics/fv/biot.py [hunk 1]
```diff
@@ -198,7 +202,7 @@ def assemble_matrix(self, g, data):
                 [A_mech, matrices_m["grad_p"] * biot_alpha],
                 [
                     matrices_m["div_d"] * biot_alpha * d_scaling,
-                    matrices_f["mass"] + dt * A_flow + matrices_f["biot_stabilization"],
+                    matrices_f["mass"] + dt * A_flow + biot_alpha * matrices_f["biot_stabilization"],
```
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/numerics/fv/biot.py]: What do you mean?

File: src/porepy/numerics/fv/biot.py [hunk 2]
```diff
@@ -386,7 +392,7 @@ def build_rhs_normals_single_dimension(dim):
         num_dir_subface = (
             bound_exclusion_mech.exclude_neu.shape[1]
             - bound_exclusion_mech.exclude_neu.shape[0]
-        ) * nd
+        )
```
Comment (keileg) [inline src/porepy/numerics/fv/biot.py]: Is this a bugfix?

File: src/porepy/numerics/fv/biot.py [hunk 3]
```diff
@@ -843,7 +850,26 @@ def assemble_rhs(self, g, data):
...
                 conditions.
         """
-        return np.zeros(self.ndof(g))
+        parameter_dictionary = data[pp.PARAMETERS][self.keyword]
+        matrix_dictionary = data[pp.DISCRETIZATION_MATRICES][self.keyword]
+
+        # For IE and constant BCs, the boundary part cancels, as the contribution from
+        # successive timesteps (n and n+1) appear on the rhs with opposite signs. For
+        # transient BCs, use the below with the appropriate version of d_bound_i.
+        d_bound_1 = parameter_dictionary["bc_values"]
```
Comment (keileg) [inline src/porepy/numerics/fv/biot.py]: Is this correct? Will not d_bound_1 == d_bound_0

File: src/porepy/numerics/fv/biot.py (hunk 2 above)
Comment (rbe051) [reply src/porepy/numerics/fv/biot.py]: Yes, bound_exlusioin.exlude_neu (and _dir, etc) changed from 1 per face to nd per face at some point (related to mpsa always getting BoundaryConditionVectorial).

File: src/porepy/numerics/fv/biot.py (hunk 3 above)
Comment (rbe051) [reply src/porepy/numerics/fv/biot.py]: I belive it is correct. If you have constant boundary conditions  d_bound_1 == d_bound_0, but this is not true for time dependent boundary conditions. Time dependent boundary conditions are not supported yet, but Ivar and I agreed it was best to still write the equation down to make it explicit

File: src/porepy/numerics/fv/biot.py (hunk 2 above)
Comment (rbe051) [reply src/porepy/numerics/fv/biot.py]: Yes, because bound_exclusion has changed from operating

File: src/porepy/numerics/fv/biot.py (hunk 3 above)
Comment (keileg) [reply src/porepy/numerics/fv/biot.py]: I should have read the comments more carefully then. Good.

File: src/porepy/numerics/fv/biot.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbe051) [reply src/porepy/numerics/fv/biot.py]: In the previous version the stabilization term did not include biot alpha.
I multiplied the stabilization term with the biot alpha.
Do you agree that the stabilization term should include a biot alpha?

**Comment by keileg** [comment]
Yes.

**Pull request merged**